database.py

```diff
@@ -40,14 +56,21 @@ def __str__(self):
 
 
 class Database:
+    """The Database object is a container for all the supported Pokemon."""
     __pokemon_list = []
     __pokemon_dictionary = {}
+    __pokemon_type_dictionary = {}
+    __POKEMON_TYPES = ('normal', 'fire', 'fighting', 'water', 'flying', 'grass', 'poison',
+                       'electric', 'ground', 'psychic', 'rock', 'ice', 'bug', 'dragon', 'ghost',
+                       'dark', 'steel', 'fairy')
```

Comment:
I mean if I understand what you mean i could do make some crazy juggling on the loading to dynamically create the arrays and keep track of the types as they appear, but I didn't think such higher complexity would help in any way shape of form. This is just as efficient and way more clear.